pyccel/parser/syntactic.py

```diff
@@ -569,7 +573,24 @@ def _visit_DefArgumentNode(self, stmt):
             return ValuedArgument(arg, value)
 
     def _visit_ReturnNode(self, stmt):
-        expr = Return(self._visit(stmt.value))
+        results = self._visit(stmt.value)
+        if not isinstance(results, (list, PythonTuple, List)):
+            results = [results]
+        assigns  = []
+        new_vars = []
+        for result in results:
+            if not isinstance(result, Symbol):
+                new_vars.append(create_variable(result))
+                new_stmt  = Assign(new_vars[-1], result)
+                new_stmt.set_fst(stmt)
+                assigns.append(new_stmt)
+            else:
+                new_vars.append(result)
+
+        if assigns:
+            expr = Return(new_vars, CodeBlock(assigns))
+        else:
+            expr = Return(new_vars)
```

Comment:
@saidctb I see you moved the treatment of multiple return values from the semantic to the syntactic stage. What is the advantage of this approach?

Reply:
because we need to have access to the results names in FunctionDef  when we visit it in semantics 
otherwise we can't  only after annotation

Reply:
That makes sense, thanks!